### PR TITLE
Establish series/0.7

### PR DESCRIPTION
**Pull request by armanbilge**
I'm targeting this at series/0.5 since that's what it's based on, but it should get its own branch :)

Mostly opening for CI now.

I updated you to the latest sbt-http4s-org and migrated the site to Laika. We'll have to figure out the versioning problem later, but seeing as right now you have no landing page at all this should be an improvement 😉 

The site will need some tweaking, but you can ~preview it as long as my workspace is running~.
~https://8000-armanbilge-http4sjdkhttp-x69u1ksax0d.ws-us30.gitpod.io/#~
https://armanbilge.github.io/http4s-jdk-http-client

### CODE REVIEW

**Comment by armanbilge** [comment]
Ok, this is actually ready 😁 

Ignore the site failure—it's only failing because it's attempting to deploy from this PR because I stupidly named the branch on my repo `series/0.7`. OTOH, it did get published from my repo:
https://armanbilge.github.io/http4s-jdk-http-client

Note that I had to disable mdoc for the websocket example, because those APIs are not yet public in http4s-client.

**Comment by rossabaker** [comment]
I wouldn't be surprised if an http4s-0.23 gets released this week... are we feeling confident in the WS implementation?  I can try to find time to examine this, but I'm not a web socket expert...

**Comment by armanbilge** [comment]
The "low-level" websocket client and model in http4s proper is so similar to what was already here that I didn't really change anything.

File: build.sbt [hunk 1]
```diff
@@ -58,145 +54,61 @@ val coreDeps = Seq(
...
+    "amesgen",
+    "Alexander Esgen",
+    "amesgen@amesgen.de",
+    url("https://github.com/amesgen")
+  ),
+  Developer(
+    "rossabaker",
+    "Ross A. Baker",
+    "ross@rossabaker.com",
+    url("https://github.com/rossabaker")
```
Comment (amesgen) [inline build.sbt:78]:
```suggestion
  tlGitHubDev("ChristopherDavenport", "Christopher Davenport"),
  tlGitHubDev("amesgen", "Alexander Esgen"),
  tlGitHubDev("rossabaker", "Ross A. Baker")
```
Feel free to also add yourself if you want ;)

File: build.sbt [hunk 2]
```diff
@@ -58,145 +54,61 @@ val coreDeps = Seq(
...
-    ),
-    githubWorkflowPublishTargetBranches := Seq(
-      RefPredicate.Equals(Ref.Branch("main")),
-      RefPredicate.StartsWith(Ref.Branch("series/")),
-      RefPredicate.StartsWith(Ref.Tag("v"))
-    )
+ThisBuild / crossScalaVersions := Seq("2.12.15", "2.13.7", "3.1.0")
+ThisBuild / scalaVersion := (ThisBuild / crossScalaVersions).value.head
+ThisBuild / tlBaseVersion := "0.7"
+ThisBuild / startYear := Some(2021)
```
Comment (amesgen) [inline build.sbt:60]: Shouldn't this maybe be
```suggestion
ThisBuild / startYear := Some(2019)
```
according to https://github.com/http4s/http4s-jdk-http-client/graphs/code-frequency?

File: build.sbt [hunk 3]
```diff
@@ -58,145 +54,61 @@ val coreDeps = Seq(
...
-    ghpagesCleanSite / includeFilter :=
-      new FileFilter {
-        def accept(f: File) =
-          f.toPath.startsWith(
-            (ghpagesRepository.value / (Paradox / siteSubdirName).value).toPath
-          )
-      } || "netlify.toml"
+    unusedCompileDependenciesFilter -= moduleFilter("org.http4s", "http4s-blaze-server"),
+    unusedCompileDependenciesFilter -= moduleFilter("org.http4s", "http4s-dsl"),
+    unusedCompileDependenciesFilter -= moduleFilter("org.scalameta", "mdoc")
```
Comment (amesgen) [inline build.sbt:109]: Maybe we just want a catch all?
```suggestion
    unusedCompileDependenciesFilter -= moduleFilter()
```

File: build.sbt (hunk 2 above)
Comment (armanbilge) [reply build.sbt:60]: I chose this to preserve your existing copyright headers, but agree about the start date :)
> Copyright 2021 http4s.org

File: build.sbt (hunk 3 above)
Comment (armanbilge) [reply build.sbt:109]: Cool, don't have much experience with this plugin 👍

File: build.sbt (hunk 2 above)
Comment (amesgen) [reply build.sbt:60]: 2021 was the year when we introduced the license header plugin 😆

Comment (amesgen) [reply build.sbt:60]: I don't think this really matters, let's just leave it as is :+1: (but feel free to update if you want).

Comment (armanbilge) [reply build.sbt:60]: Oh, just about to push :)

File: build.sbt (hunk 1 above)
Comment (amesgen) [reply build.sbt]: Sry, ust noted that I missed one closing `)` here...

**Comment by armanbilge** [comment]
@amesgen you should make a series/0.7 branch on the repo that I can target this to.

**Comment by armanbilge** [comment]
Also, I just released sbt-http4s-org 0.12.1 with the latest stuff, we should update to that before merging.

**Comment by amesgen** [comment]
> @amesgen you should make a series/0.7 branch on the repo that I can target this to.

Done :+1:

**Comment by armanbilge** [comment]
@amesgen should be good-to-go!

**Comment by amesgen** [comment]
AFAIUI his should be cherry-picked on top of `main`, will do this now.

**Pull request merged**